portfolio/src/main/webapp/buy_rent_logic.js

```diff
@@ -0,0 +1,29 @@
+function calcBuyRent()
+{
+    const home_price = document.getElementById('home_cost').value;
+    const rent_price = document.getElementById('rent_cost').value;
+    const num_years = document.getElementById('time').value;
+
+    var year_rent = rent_price * 12;
+    var total_rent = year_rent;
+    var year = 1;
+
+    while (total_rent < home_price)
+    {
+        total_rent = total_rent + year_rent;
+        year = year + 1;
+    }
+    
+    const result_text = document.getElementById('calcResult');
+    const advice_text = document.getElementById('calcAdvice');
+    result_text.innerText = 'After ' + year + ' years, you would have paid the same amount for rent and a home!';
+    
+    if (num_years >= year)
```

Comment:
I think you should use more descriptive variable names here. It looks like `year` is the number of years before breakeven, and `num_years` is the number of years they expect to live in the location?